### PR TITLE
Factor out search

### PR DESCRIPTION
**Pull request by eddieantonio**
I have been uneasy about the god class that became of `Wordform`. It just had too much junk in it! So, I placed A LOT OF THAT into a new file — search.py. It has stuff related to searching!

This is how a search is performed now:

(**EDIT**: I ended up reverting the change to the API methods ¯\\\_(ツ)\_/¯  )

```python
search_results = Wordform.search("nêhiyawêwin")
search_results = Wordform.search("English")
```

Now `models.py` has a more favourable ratio of Django stuff, like defining model fields, than our search logic.

---

@Madoshakalaka feel free to reject this refactor if it seems too drastic!

### CODE REVIEW

File: CreeDictionary/API/models.py [hunk 1]
```diff
@@ -349,485 +226,12 @@ def save(self, *args, **kwargs):
...
-    """
+    @classmethod
+    def search(cls, query: str, **constraints) -> SortedSet["SearchResult"]:
+        from .search import WordformSearch
 
-    # MatchedCree are inflections
-    cree_results: Set[CreeResult]
-    english_results: Set[EnglishResult]
+        search = WordformSearch(query, constraints)
+        return search.perform()
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py:234]: Delegate entirely to the new search module.

File: CreeDictionary/API/models.py [hunk 2]
```diff
@@ -1,157 +1,34 @@
...
 
-def replace_user_friendly_tags(fst_tags: List[FSTTag]) -> List[Label]:
-    """ replace fst-tags to cute ones"""
-    return LABELS.english.get_full_relabelling(fst_tags)
+# Avoid a runtime circular-dependency;
+# without this line,
+#  - model.py imports search.py; and,
+#  - search.py imports model.py 💥
+if typing.TYPE_CHECKING:
+    from .search import SearchResult
```
Comment (Madoshakalaka) [inline CreeDictionary/API/models.py:28]: interesting!

File: CreeDictionary/API/search.py [hunk 3]
```diff
@@ -0,0 +1,606 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (Madoshakalaka) [reply CreeDictionary/API/search.py:1]: nice!

File: CreeDictionary/API/utils.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from sortedcontainers import SortedSet
+
+
+class SortedSetWithExtend(SortedSet):
+    def extend(self, items):
+        for it in items:
+            self.add(it)
```
[the remark this answers is not in the document]
Comment (Madoshakalaka) [reply CreeDictionary/API/utils.py:7]: They do have magic method for set union.
```
>>> the_set = SortedSet([1,2,3])
>>> the_set
SortedSet([1, 2, 3])
>>> that_set = SortedSet([4,5,6])
>>> the_set | that_set
SortedSet([1, 2, 3, 4, 5, 6])
```

[the remark this answers is not in the document]
Comment (eddieantonio) [reply CreeDictionary/API/utils.py:7]: Thanks for reminding me about this! Makes the code much simpler!

**Pull request merged**